receiver/prometheusreceiver/config.go

```diff
@@ -71,7 +72,77 @@ func (cfg *PromConfig) ContainsScrapeConfigs() bool {
 	return cfg != nil && (len(cfg.ScrapeConfigs) > 0 || len(cfg.ScrapeConfigFiles) > 0)
 }
 
+func (cfg *PromConfig) ContainsSecrets() bool {
+	if cfg == nil {
+		return false
+	}
+
+	return findSecretsRecursive(reflect.ValueOf(cfg))
+}
+
+// findSecretsRecursive recursively traverses the fields of a reflect.Value v.
+// It returns true if a non-empty commonconfig.Secret is found.
+func findSecretsRecursive(v reflect.Value) bool {
+	// 1. Handle pointers and interfaces: get the actual underlying value.
+	for v.Kind() == reflect.Ptr || v.Kind() == reflect.Interface {
+		if v.IsNil() {
+			return false
+		}
+		v = v.Elem()
+	}
+
+	// 2. Handle slices and arrays: iterate over elements and recurse.
+	if v.Kind() == reflect.Slice || v.Kind() == reflect.Array {
+		for i := 0; i < v.Len(); i++ {
+			if findSecretsRecursive(v.Index(i)) {
+				return true
+			}
+		}
+		return false
+	}
+
+	// 3. We only care about structs for field iteration from this point.
+	if v.Kind() != reflect.Struct {
+		return false
+	}
```

Comment:
There's technically a bug here, if a field is an array of `Secret`. Will fix this.